src/models/Location.java

```diff
@@ -142,6 +142,11 @@ public static Location createInitialLocation(
         return create(name, invariant, true, isUrgent, isUniversal, isInconsistent);
     }
 
+    public static Location createFromState(State state, List<Clock> clocks) {
+        Location location = state.getLocation();
+        return location.copy();
+    }
```

Comment:
What is the purpose of `List<Clock> clocks` parameter in this function? I don't see it explicitly being used.

Reply:
I am not entirely sure. I think it was originally used to create a state with only a specific set of clocks. I will have to ask Florian.